### PR TITLE
Raise an error when updating a runtime currently in use.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2091" title="DX-2091" target="_blank">StoryDX-2091</a>  We ensure the runtime is not in use before updating it
  </td></table>
  <br />

### CODE REVIEW

**Comment by mitchell-as** [comment]
Test failures are not due to this PR.

File: pkg/platform/runtime/setup/setup.go [hunk 1]
```diff
@@ -220,6 +235,35 @@ func (s *Setup) Update() (rerr error) {
 	return nil
 }
 
+func (s *Setup) getProcessesInUse() map[string]int32 {
```
Comment (MDrakos) [inline pkg/platform/runtime/setup/setup.go:238]: I think it makes more sense for this function to be part of the `osutils` package and we can supply it the directory to compare. Also we could have os-specific versions of these functions to reduce the if statements here.

**Review by MDrakos — CHANGES_REQUESTED** [review]

File: pkg/platform/runtime/setup/setup.go (hunk 1 above)
Comment (mitchell-as) [reply pkg/platform/runtime/setup/setup.go]: Okay, I moved it into the `osutils` package. However, when I tried splitting it up into OS-specific functions, I ended up with too much duplication. I left it as a single function.

**Review by MDrakos — APPROVED** [review]
Looks good, is there an easy way for us to introduce an integration test for this? If it's too much it can be a follow-up story. Just a nice to have right now.

**Comment by mitchell-as** [comment]
https://activestatef.atlassian.net/browse/DX-2331

**Pull request merged**